File: solutions/csharp/log-levels/1/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        int i = logLine.IndexOf(" ");
        string msg = logLine.Substring(i + 1);
        return msg.Trim();
    }

    public static string LogLevel(string logLine)
    {
        int i = logLine.IndexOf("]");
        string lvl = logLine.Substring(1, i - 1);
        return lvl.ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}

File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message(logLine: string): string = (Array.get (logLine.Split(':')) 1).Trim()

let logLevel(logLine: string): string = match logLine with
                                        | line when line.StartsWith("[ERROR]") -> "error"
                                        | line when line.StartsWith("[WARNING]") -> "warning"
                                        | line when line.StartsWith("[INFO]") -> "info"
                                        | _ -> "default"

let reformat(logLine: string): string = message(logLine) + " (" + logLevel(logLine) + ")"
